var/spack/repos/builtin/packages/dihydrogen/package.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Dihydrogen(CMakePackage):
+    """DiHydrogen is the second version of the Hydrogen fork of the
+       well-known distributed linear algebra library,
+       Elemental. DiHydrogen aims to be a basic distributed
+       multilinear algebra interface with a particular emphasis on the
+       needs of the distributed machine learning effort, LBANN."""
+
+    homepage = "https://github.com/LLNL/DiHydrogen.git"
+    url      = "https://github.com/LLNL/DiHydrogen.git"
+    git      = "https://github.com/LLNL/DiHydrogen.git"
+
+    maintainers = ['bvanessen']
+
+    version('master', branch='master')
+
+    variant('al', default=True,
+            description='Builds with Aluminum communication library')
+    variant('cuda', default=False,
+            description='Builds with support for GPUs via CUDA')
+    variant('developer', default=False,
+            description='Enable extra warnings and force tests to be enabled.')
+    variant('half', default=False,
+            description='Enable FP16 support on the CPU. Requires the Half library.')
+    variant('legacy', default=False,
+            description='Enable the legacy DistConv code branch.')
+    variant('nvshmem', default=False,
+            description='Builds with support for NVSHMEM')
+    variant('openmp', default=False,
+            description='Enable CPU acceleration with OpenMP threads.')
+    variant('rocm', default=False,
+            description='Search for and enable ROCm/HIP language features in DiHydrogen.')
+    variant('shared', default=True,
+            description='Enables the build of shared libraries')
+    variant('test', default=False,
+            description='Builds test suite')
```

Comment:
This variant should probably be removed if it doesn't affect what gets installed. If you're just trying to build the test suite to test during build-time, you can check whether `self.run_tests` is True.